garage/tf/algos/vpg.py

```diff
@@ -9,6 +9,7 @@ class VPG(NPO):
     """
 
     def __init__(self, optimizer=None, optimizer_args=None, **kwargs):
+        self.quick_init(locals())
```

Comment:
This should not be called here, since it's called in `VPG.__init__`.

Reply:
are we sure? i think have encountered behavior with `Serializable` requiring it be called in the constructor of every type you'd like to pickle (including child classes).

if a child class adds a constructor argument not present in the parent, i don't think Serializable will catch that in the parent. Original rllab code made a habit of peppering *args and **kwargs all over all constructors to catch this flat (I speculate). You can imagine that neither choice is a particularly good one.

Reply:
To be clear, it doesn't break anything to call `quick_init()` here. Instead, this call silently does nothing, since the *last* call (inside `NPO.__init__`) to `quick_init()` will override it. Read the source of `quick_init()` for why.

The correct thing to do is probably to call `quick_init()` at the *end* of these `__init__` functions, but that breaks the `quick_init()` at start of `__init__` rule, so it's also fragile.